File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;

        unordered_map<int, int> mp;

        for(int i=0; i < nums.size(); i++) {
            int b = target - nums[i];
            if(mp.count(b) == 1) {
                res.push_back(mp[b]);
                res.push_back((i));
                return res;
            }
            mp[nums[i]] = i;
        }
        return {-1, -1};
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-2.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []

        def dfs(i: int, cur_sum: int, arr: List[int]) -> None:
            if(cur_sum == target):
                res.append(arr.copy())
                # res.add(tuple(arr.copy()))
                return
            
            if(cur_sum > target):
                return
            
            if(i >= len(nums)):
                return
            
            #Dont pick, go next
            dfs(i + 1, cur_sum, arr)

            arr.append(nums[i])

            #Pick, stay
            dfs(i, cur_sum + nums[i], arr)

            arr.pop()

            return

        dfs(0, 0, [])

        # final = []

        # for key in res:
        #     final.append(list(key))

        return res

